crates/solver-types/src/api.rs

```diff
@@ -406,6 +408,237 @@ impl From<GetOrderError> for APIError {
 	}
 }
 
+/// A tuple combining Quote and signature for intent request conversion
+#[cfg(feature = "standard")]
+pub struct QuoteWithSignature<'a> {
+	pub quote: &'a Quote,
+	pub signature: &'a str,
+}
+
+#[cfg(feature = "standard")]
+impl<'a> From<(&'a Quote, &'a str)> for QuoteWithSignature<'a> {
+	fn from((quote, signature): (&'a Quote, &'a str)) -> Self {
+		Self { quote, signature }
+	}
+}
+
+#[cfg(feature = "standard")]
+impl TryFrom<QuoteWithSignature<'_>> for serde_json::Value {
+	type Error = Box<dyn std::error::Error>;
+
+	fn try_from(quote_with_sig: QuoteWithSignature<'_>) -> Result<Self, Self::Error> {
+		use crate::standards::eip7683::standard::{SolMandateOutput, StandardOrder};
+		use crate::standards::eip7930::InteropAddress;
+		use alloy_primitives::{Address, U256};
+		use alloy_sol_types::SolType;
+
+		// Extract user address from the first available input
+		let user_str = &quote_with_sig
+			.quote
+			.details
+			.available_inputs
+			.first()
+			.ok_or("Quote must have at least one available input")?
+			.user;
+		let interop_address = InteropAddress::from_hex(&user_str.to_string())?;
+		let user_address = interop_address.ethereum_address()?;
+
+		// Extract order data from quote
+		let quote_order = quote_with_sig
+			.quote
+			.orders
+			.first()
+			.ok_or("Quote must contain at least one order")?;
+		let message_data = quote_order
+			.message
+			.as_object()
+			.ok_or("Invalid EIP-712 message structure")?;
+		let eip712_data = message_data
+			.get("eip712")
+			.and_then(|e| e.as_object())
+			.ok_or("Missing 'eip712' object in message")?;
+
+		// Extract nonce
+		let nonce_str = eip712_data
+			.get("nonce")
+			.and_then(|n| n.as_str())
+			.ok_or("Missing nonce in EIP-712 data")?;
+		let nonce = U256::from_str_radix(nonce_str, 10)?;
+
+		// Extract witness data
+		let witness = eip712_data
+			.get("witness")
+			.and_then(|w| w.as_object())
+			.ok_or("Missing 'witness' object in EIP-712 message")?;
+
+		// Get origin chain ID
+		let origin_chain_id = U256::from(interop_address.ethereum_chain_id()?);
+
+		// Extract timing data
+		let expires = witness
+			.get("expires")
+			.and_then(|e| e.as_u64())
+			.unwrap_or(quote_with_sig.quote.valid_until.unwrap_or(0)) as u32;
+		let fill_deadline = expires;
+
+		// Extract input oracle
+		let input_oracle_str = witness
+			.get("inputOracle")
+			.and_then(|o| o.as_str())
+			.unwrap_or("0x0000000000000000000000000000000000000000");
```

Comment:
Let's handle this also as error instead of fallback. It should actually fail anyway because we wouldn't support a route with ZERO address oracle..